packages/experience/src/pages/SocialSignInWebCallback/use-social-sign-in-listener.ts

```diff
@@ -117,17 +127,26 @@ const useSocialSignInListener = (connectorId: string) => {
 
   const verifySocialCallbackData = useCallback(
     async (connectorId: string, data: Record<string, unknown>) => {
-      // When the callback is called from Google One Tap, the interaction event was not set yet.
-      if (data[GoogleConnector.oneTapParams.csrfToken]) {
+      const isGoogleOneTap = isGoogleOneTapChecker(data);
+      // Check for external Google One Tap credentials from extraParams
+      if (isGoogleOneTap) {
+        // External Google One Tap flow - initialize interaction for external scenario
         await asyncInitInteraction(InteractionEvent.SignIn);
       }
 
+      const { [ExtraParamsKey.GoogleOneTapCredential]: externalCredential, ...rest } = data;
       const [error, result] = await verifySocial(connectorId, {
         verificationId: verificationIdRef.current,
         connectorData: {
           // For validation use only
           redirectUri: `${window.location.origin}/callback/${connectorId}`,
-          ...data,
+          // Not using `conditional` here to make type inference work.
+          ...(externalCredential && typeof externalCredential === 'string'
+            ? {
```

Comment:
A little bit confused here, can we simply pass the data to the server side?

Reply:
Any chance to align the naming convention?

Reply:
the native param name for GOT is `credential`, which is too general. It is not a good idea to use this param name for external website GOT ID token.